### DISCUSSION TITLE
foundryup-zksync not working as the installation method for foundry-zksync || libusb trouble

### DISCUSSION THREAD

**Discussion by Stachelabs** [Cyfrin/foundry-full-course-cu#2740] [Q&A]
Hello, 

thanks for helping me, while foundryup works pretty well I have trouble with froundry for the zksync version

either the installation process or the ```froundryup-zksync``` process end up with an error related to libusb

after searching on this repo and on different ressources it appears some have either installed Xcode, either run ```brew install libusb```

I still have the issue 

```
dyld[3542]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <5E7A515C-C8FC-37C7-A150-07D1D38A0101> /Users/mo7tr/.foundry/bin/forge
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file)
foundryup-zksync: command failed: /Users/mo7tr/.foundry/bin/forge --version
foundryup-zksync: installed - 
dyld[3545]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <07ABDB06-6808-394B-8EDF-EB35474100B8> /Users/mo7tr/.foundry/bin/cast
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file)
foundryup-zksync: command failed: /Users/mo7tr/.foundry/bin/cast --version
foundryup-zksync: installed - 
foundryup-zksync: done!
Cleaning up installation artifacts...
Cleanup completed.
Installation completed successfully!
Verifying installation...
dyld[3548]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <5E7A515C-C8FC-37C7-A150-07D1D38A0101> /Users/mo7tr/.foundry/bin/forge
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file)
Installation verification failed. Forge is not properly installed.
```

AI didn't help much

Thanks for your help

**Comment by EngrPips** [comment]
Did you have `libusb` installed?

**Reply by Stachelabs** [reply]
yes libusb is installed and up to date! don't matter, i'll move on without zksync, it seems that a lot of persons got in trouble with that. Thanks anyway

**Reply by EngrPips** [reply]
Yeah, Working with `zksync` is not very smooth right now.

**Reply by ericchelini** [reply]
When will ZKsync get smoother?

**Reply by EngrPips** [reply]
Soon, I hope. ZKSync has many fundamental differences from it's L1 blockchain, and it will take some time to make things start working seamlessly.

**Comment by wharika18** [comment]
I too encountered the same err.. but this command ```brew install libusb``` did the magic .. though u may have the libusb already installed, try this once , any issues in the installation may be rectified..
1) brew install libusb
2) brew link libusb
later check forge --version , to verify if the zksync is installed or not 
Hope this helps ..

**Reply by EngrPips** [reply]
Thanks for this heads-up

**Reply by nikolov9996** [reply]
It worked for me :)

**Reply by EngrPips** [reply]
Awesome.
